### PR TITLE
Raise exception message to prevent segfault

### PR DESCRIPTION
**Pull request by blaenk**
On macOS, using the built-in system libgssapi leads to a segfault. This attempts to detect if the non-homebrew library is being used and if so raises an exception instructing the user to install the `krb5` homebrew package and add it to their PATH. This exception is raised to avoid continuing with the use of the system library, since that would result in a segfault.

I can't think of a foolproof way of detecting the use of the macOS built-in libgssapi library, beyond checking some SHA hash or other kind of fingerprint, such as of its exported symbols, and I'm not sure if that would even be future-proof or even correct. So for now this enforces that the homebrew version is being used, as opposed to preventing the built-in one from being used. So it's more strict than it probably needs to be.

It does this by ensuring that the library path has the Homebrew Cellar path as an ancestor.

Closes Factual/data-projects#1842

### CODE REVIEW

File: lib/gssapi/lib_gssapi_loader.rb [hunk 1]
```diff
@@ -26,16 +26,24 @@ def self.load_mit
         gssapi_lib = 'libgssapi_krb5.so.2'
         ffi_lib gssapi_lib, FFI::Library::LIBC
       when /darwin/
-        prefix = begin
-          stdout, _stderr, status = Open3.capture3("krb5-config --prefix")
-          stdout.chomp!
+        exception_message = <<BREW
+The gssapi-factual gem on macOS requires the krb5 package from Homebrew. `brew install krb5` and ensure it's in your PATH"
```
Comment (asteadyfactual) [inline lib/gssapi/lib_gssapi_loader.rb]: total nitpick, but it looks like you have a `"` at the end, but i don't see a matching one. should still work with the heredoc format, but not sure if you wanted to specifically quote a part of the message

**Review by asteadyfactual — APPROVED** [review]
assuming you were able to run this and saw that it works as expected, this lgtm

**Comment by blaenk** [comment]
Yup, thanks!

File: lib/gssapi/lib_gssapi_loader.rb (hunk 1 above)
Comment (blaenk) [reply lib/gssapi/lib_gssapi_loader.rb]: Woah you're absolutely right, thanks. It got left over from when I converted it to a heredoc.

**Pull request merged**